File: src/main/java/hu/elte/easybet/model/Bet.java
package hu.elte.easybet.model;

import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

import com.fasterxml.jackson.annotation.JsonIgnore;

@Entity
@Table(name = "bets")
@Data
@AllArgsConstructor
@NoArgsConstructor
//@EqualsAndHashCode(callSuper = true)
public class Bet extends BaseEntity{

    private Boolean active;
    
    private Boolean winner;
    
    @Column(nullable = false)
    private double stake;

    @Column(nullable = false)
    private double winnableStake;
    
    @ManyToOne(optional=false)
    @JoinColumn(name="users_id", referencedColumnName="id")
    private User user;
    
    @ManyToOne(optional=false)
    @JoinColumn(name = "events_id", referencedColumnName = "id")
    private Event connectedEvent;

}

File: frontend/src/app/model/Bet.ts
import { User } from './User';
import { Event } from './Event';

export class Bet {
    id: number;
    active: boolean;
    winner: boolean;
    stake: number;
    winnableStake: number;
    user: User;
    connectedEvent: Event;

    constructor(active?: boolean, winner?: boolean, stake?: number, winnableStake?: number, user?: User, connectedEvent?: Event, id?: number) {
        this.active = active;
        this.winner = winner;
        this.stake = stake;
        this.winnableStake = winnableStake;
        this.user = user;
        this.connectedEvent = connectedEvent;
        this.id = id;
    }
}
